# Contributor: Antoine Martin (ayakael) <dev@ayakael.net>
# Maintainer: Antoine Martin (ayakael) <dev@ayakael.net>
pkgname=py3-swagger-spec-validator
_pyname=swagger-spec-validator
pkgver=3.0.3
pkgrel=3
arch="noarch"
pkgdesc="Validation of Swagger specifications"
url="https://pypi.python.org/project/swagger-spec-validator"
license="Apache-2.0"
depends="
	py3-jsonschema
	py3-yaml
	py3-typing-extensions
	"
makedepends="
	py3-setuptools
	py3-gpep517
	py3-wheel
	"
checkdepends="
	py3-pytest
"
source="$pkgname-$pkgver-gh.tar.gz::https://github.com/Yelp/swagger_spec_validator/archive/refs/tags/v$pkgver.tar.gz"
builddir="$srcdir"/swagger_spec_validator-$pkgver
subpackages="$pkgname-pyc"

build() {
	gpep517 build-wheel \
		--wheel-dir .dist \
		--output-fd 3 3>&1 >&2
}

check() {
	python3 -m venv --clear --without-pip --system-site-packages .testenv
	.testenv/bin/python3 -m installer .dist/*.whl
	# test_raise_SwaggerValidationError_on_urlopen_error: urlopen error [Errno -5] Name has no usable address>', URLError(gaierror(-5, 'Name has no usable address'
	.testenv/bin/python3 -m pytest -v -k 'not test_raise_SwaggerValidationError_on_urlopen_error'
}

package() {
	python3 -m installer -d "$pkgdir" \
		.dist/*.whl
}

sha512sums="
977a6320ce629d102c7c608249d220996aba852d1c0c464be447b6b5a7aa7bb2a61023bc967a41c561a12c4620306c6a077605c4b34b828540d0ee8926d59a94  py3-swagger-spec-validator-3.0.3-gh.tar.gz
"
